aplib.net-demo/Assets/Scripts/GameManager.cs

```diff
@@ -0,0 +1,30 @@
+using UnityEngine;
+
+public class GameManager : MonoBehaviour
+{
+  private static GameManager _instance;
+
+  public static GameManager Instance
+  {
+    get
+    {
+      if (_instance == null)
+      {
+        Debug.LogError("Game Manager is null, first awake it!");
+      }
+      
+      return _instance;
+    }
+  }
+
+  private void Awake()
+  {
+        // Setup singleton
+        if (_instance != null && _instance != this) {
+            Destroy(gameObject);
+        } else {
+            _instance = this;
+        }
+        DontDestroyOnLoad(gameObject);
```

Comment:
Indentation is weird because of my suggestion. I'm not sure how to indent it correctly in a suggestion.